rskj-core/src/test/java/co/rsk/peg/StandardMultisigFederationTest.java

```diff
@@ -121,52 +121,20 @@ void redeemScript() {
     }
 
     @Test
-    void P2SHScript() {
-        final List<Integer> calls = new ArrayList<>();
-        try (MockedStatic<ScriptBuilder> scriptBuilderMocked = mockStatic(ScriptBuilder.class)) {
-            scriptBuilderMocked.when(() -> ScriptBuilder.createP2SHOutputScript(any(int.class), any(List.class))).thenAnswer((invocationOnMock) -> {
-                calls.add(0);
-                int numberOfSignaturesRequired = invocationOnMock.<Integer>getArgument(0);
-                List<BtcECKey> publicKeys = invocationOnMock.getArgument(1);
-                Assertions.assertEquals(4, numberOfSignaturesRequired);
-                Assertions.assertEquals(6, publicKeys.size());
-                for (int i = 0; i < sortedPublicKeys.size(); i++) {
-                    Assertions.assertArrayEquals(sortedPublicKeys.get(i).getPubKey(),
-                            publicKeys.get(i).getPubKey());
-                }
-                return new Script(new byte[]{(byte) 0xaa});
-            });
-            Assertions.assertArrayEquals(federation.getP2SHScript().getProgram(), new byte[]{(byte) 0xaa});
-            // Make sure the script creation happens only once
-            Assertions.assertEquals(1, calls.size());
-        }
+    void testArrayEquals_P2SHScript() {
+        ScriptBuilder p2shScriptBuilder = new ScriptBuilder();
+        p2shScriptBuilder.op(ScriptOpCodes.OP_HASH160);
+        p2shScriptBuilder.data(Hex.decode("57f76bf3ab818811c740929ac7a5e3ef8c7a34b9"));
+        p2shScriptBuilder.op(ScriptOpCodes.OP_EQUAL);
+
+        Script p2shScript = p2shScriptBuilder.build();
+
+        Assertions.assertArrayEquals(federation.getP2SHScript().getProgram(), p2shScript.getProgram());
```

Comment:
```suggestion
        Assertions.assertEquals(federation.getP2SHScript(), p2shScript);
```
That should do it, right?

Reply:
Did you try this?